app/connectors/ApiConnector.scala

```diff
@@ -35,6 +37,28 @@ class ApiConnector @Inject() (httpClient: HttpClient, appConfig: AppConfig)(impl
     HeaderNames.CONTENT_TYPE -> "application/xml"
   )
 
+  def getDepartures(queryParams: Seq[(String, String)] = Seq.empty)(implicit hc: HeaderCarrier): Future[Option[Departures]] = {
+    val url = s"${appConfig.apiUrl}/movements/departures"
+
+    val headers = hc.withExtraHeaders(("Accept", "application/vnd.hmrc.2.0+json"))
+
+    httpClient
+      .GET[HttpResponse](url, queryParams)(rawHttpResponseHttpReads, headers, ec)
```

Comment:
All looks fine to me. Out of curiosity, does everything still pass if you remove the `rawHttpResponseHttpReads` and replace it with `implicitly`?

Reply:
Messes up one test

- must return empty Departures when 404 is returned *** FAILED ***
[info]     None was not equal to Some(Departures(List())) (ApiConnectorSpec.scala:386)

I think it must use a different HttpReader where as the rawHttpResponseHttpReads converts it into a None if empty I believe

Reply:
So I suspect what it actually is is a 404 falls into the recover block. The raw response reader just returns the raw response regardless of the status code, so it's aptly named!